addon-test-support/logger.ts

```diff
@@ -0,0 +1,186 @@
+import axeCore, { AxeResults, NodeResult, RelatedNode, Result } from 'axe-core';
+
+/**
+ * This file is heavily borrowed from https://github.com/dequelabs/react-axe
+ *
+ * An issue (https://github.com/dequelabs/react-axe/issues/180) has been opened which
+ * suggests extracting out the common pieces of axe formatting to a separate repository,
+ * which would allow a lot of these functions to be removed.
+ */
+
+// @reference axe-core https://github.com/dequelabs/axe-core/blob/develop/lib/core/base/audit.js
+type AxeCoreNodeResultKey = 'any' | 'all' | 'none';
+
+interface AxeWithAudit {
+  _audit: {
+    data: {
+      failureSummaries: {
+        any: {
+          failureMessage: (args: string[]) => string;
+        };
+        all: {
+          failureMessage: (args: string[]) => string;
+        };
+        none: {
+          failureMessage: (args: string[]) => string;
+        };
+      };
+    };
+  };
+}
+
+// contrasted against Chrome default color of #ffffff
+const lightTheme = {
+  serious: '#d93251',
+  minor: '#d24700',
+  text: 'black',
+};
+
+// contrasted against Safari dark mode color of #535353
+const darkTheme = {
+  serious: '#ffb3b3',
+  minor: '#ffd500',
+  text: 'white',
+};
+
+const theme =
+  window.matchMedia && window.matchMedia('(prefers-color-scheme: dark)').matches
+    ? darkTheme
+    : lightTheme;
+
+const boldCourier = 'font-weight:bold;font-family:Courier;';
+const critical = `color:${theme.serious};font-weight:bold;`;
+const serious = `color:${theme.serious};font-weight:normal;`;
+const moderate = `color:${theme.minor};font-weight:bold;`;
+const minor = `color:${theme.minor};font-weight:normal;`;
+const defaultReset = `font-color:${theme.text};font-weight:normal;`;
+
+const testSuiteResults: Result[][] = [];
+const cache: { [key: string]: string } = {};
+
+function deduplicateViolations(results: AxeResults) {
+  return results.violations.filter((result) => {
+    result.nodes = result.nodes.filter((node) => {
+      const key: string = node.target.toString() + result.id;
+      const retVal = !cache[key];
+      cache[key] = key;
+      return retVal;
+    });
+    return !!result.nodes.length;
+  });
+}
+
+/**
+ * Log the axe result node to the console
+ * @param {NodeResult} node
+ * @param {Function} logFn console log function to use (error, warn, log, etc.)
+ */
+function logElement(
+  node: NodeResult | RelatedNode,
+  logFn: (...args: any[]) => void
+): void {
+  const el = document.querySelector(node.target.toString());
+  if (!el) {
+    logFn('Selector: %c%s', boldCourier, node.target.toString());
+  } else {
+    logFn('Element: %o', el);
+  }
+}
+
+/**
+ * Log the axe result node html tot he console
```

Comment:
```suggestion
 * Log the axe result node html to the console
```

Reply:
Ha. I guess they need to fix this in their repo too :)